### PR TITLE
wip benchmarks

### PR DESCRIPTION
**Pull request by t0yv0**
# Description

Prototyping. Still a bit of work to do tidy up. Thanks @AaronFriel and @RobbieMcKinstry for hyperfine suggestion. Thanks Robbie for researching the benchmark setup script I'm copying it verbatim from your instructions.

```
$ ./script/benchark.sh mini-js

Created stack 'benchmark'
Benchmark 1: control pulumi:v3.36.0 command:'pulumi up -y'
  Time (mean ± σ):      4.473 s ±  0.066 s    [User: 0.752 s, System: 0.639 s]
  Range (min … max):    4.335 s …  4.577 s    10 runs

Benchmark 1: test pulumi:3.37.0-alpha.1658432973+98555086 command:'pulumi up -y'
  Time (mean ± σ):      4.092 s ±  0.189 s    [User: 0.716 s, System: 0.577 s]
  Range (min … max):    3.837 s …  4.537 s    10 runs
```
Fixes # (issue)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: tests/benchmarks/mini-js/Pulumi.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: mini-js
+runtime:
+  name: nodejs
+  options:
+    typescript: false
```
[the remark this answers is not in the document]
Comment (t0yv0) [reply tests/benchmarks/mini-js/Pulumi.yaml:5]: It doesn't, no! This was a surprising find for me. We could consider that autodetect, WDYT?

I was debating this in my head but opted for just changing templates for now https://github.com/pulumi/templates/pull/312

This means old projects will not get a speed up, only new projects.

[the remark this answers is not in the document]
Comment (AaronFriel) [reply tests/benchmarks/mini-js/Pulumi.yaml:5]: I'm not sure if we'd want to do that autodetection as handling edge cases could be messy:

* In a monorepo, node_modules could contain plain TS (`yarn link`/lerna/etc.) files, a way of skipping transpile steps
* Even in a "plain" project folder, the root file could be index.js but it could import a .ts file.

I think to reliably detect "ts", we'd have to implement Node module resolution & import. Easier to allow opt out, I think?

[the remark this answers is not in the document]
Comment (t0yv0) [reply tests/benchmarks/mini-js/Pulumi.yaml:5]: Did not know that JS could import typescript. Yes let's not pursue auto-detection if we do not know a reliable way to do it. Thanks!
